File: dodam-application/dodam-rest-api/src/main/java/b1nd/dodam/restapi/support/file/FileController.java
package b1nd.dodam.restapi.support.file;

import b1nd.dodam.restapi.support.data.ResponseData;
import b1nd.dodam.simple.storage.service.client.SimpleStorageServiceClient;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequestMapping(value = "/upload")
@RequiredArgsConstructor
public class FileController {

    private final SimpleStorageServiceClient simpleStorageServiceClient;

    @PostMapping
    public ResponseData<String> uploadFile(@RequestParam("file") MultipartFile multipartFile) {
        return ResponseData.created("파일 업로드 성공", simpleStorageServiceClient.uploadFile(multipartFile));
    }

}

File: services/service-file/src/main/kotlin/com/b1nd/dodamdodam/file/presentation/FileController.kt
package com.b1nd.dodamdodam.file.presentation

import com.b1nd.dodamdodam.core.security.annotation.authentication.UserAccess
import com.b1nd.dodamdodam.file.application.FileUseCase
import com.b1nd.dodamdodam.file.domain.enumeration.FileType
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.multipart.MultipartFile

@RestController
class FileController(
    private val fileUseCase: FileUseCase,
) {
    @UserAccess
    @PostMapping("/upload", consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun upload(
        @RequestParam("file") file: MultipartFile,
        @RequestParam("allowType", required = false) allowType: FileType?,
        @RequestParam("width", required = false) width: Int?,
        @RequestParam("height", required = false) height: Int?,
    ) = fileUseCase.upload(file, allowType, width, height)
}
